server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/ElementaryNodes.java

```diff
@@ -0,0 +1,130 @@
+ * Contributors:
+ *    Broadcom, Inc. - initial API and implementation
+ *
+ */
+
+package org.eclipse.lsp.cobol.core.model.tree.variables;
+
+import com.google.common.collect.ImmutableList;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.lsp.cobol.core.messages.MessageTemplate;
+import org.eclipse.lsp.cobol.core.model.ErrorSeverity;
+import org.eclipse.lsp.cobol.core.model.Locality;
+import org.eclipse.lsp.cobol.core.model.ResultWithErrors;
+import org.eclipse.lsp.cobol.core.model.SyntaxError;
+import org.eclipse.lsp.cobol.core.model.variables.UsageFormat;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+
+import static org.eclipse.lsp.cobol.core.model.tree.variables.EffectiveDataType.UNDETERMINED;
+import static org.eclipse.lsp.cobol.core.model.tree.variables.VariableDefinitionUtil.FALSE_RESULT_WITH_NO_ERRORS;
+
+/**
+ * Abstract class for all elementary variables. These variables actually has space allocations and
+ * are not logical.
+ */
+@Getter
+@ToString(callSuper = true)
+@EqualsAndHashCode(callSuper = true)
+abstract class ElementaryNodes extends VariableWithLevelNode implements IUsageClause {
+  private static final String SEMANTICS_NO_PIC_CLAUSE = "semantics.noPicClause";
+  private static final String SEMANTIC_IMPROPER_USE_BLANK_WHEN_ZERO_AND_SIGN_CLAUSE =
+      "semantic.improperUseBlankWhenZeroAndSignClause";
+  private static final String BLANK_WHEN_ZERO = "BLANK WHEN ZERO";
+  private static final String SIGN_CLAUSE = "SIGN CLAUSE";
+  private final boolean isBlankWhenZeroPresent;
+  private final boolean isSignClausePresent;
+  EffectiveDataType effectiveDataType;
+  String picClause;
+  @Getter UsageFormat usageFormat;
+
+  protected ElementaryNodes(
+      Locality location,
+      int level,
+      String name,
+      VariableType variableType,
+      boolean global,
+      boolean isBlankWhenZeroPresent,
+      boolean isSignClausePresent,
+      String picClause,
+      UsageFormat usageFormat) {
+    super(location, level, name, variableType, global);
+    this.isBlankWhenZeroPresent = isBlankWhenZeroPresent;
+    this.isSignClausePresent = isSignClausePresent;
+    this.picClause = picClause;
+    this.usageFormat = usageFormat;
+    this.effectiveDataType = setEffectiveDataType();
+  }
+
+  private EffectiveDataType setEffectiveDataType() {
+    return Arrays.stream(EffectiveDataType.values())
+        .filter(type -> type.checkDataType(this))
+        .findFirst()
+        .orElse(UNDETERMINED);
+  }
+
+  /**
+   * Validate proper use of PIC and USAGE clause for elementary nodes
+   *
+   * @return SyntaxError
+   */
+  protected List<SyntaxError> validate() {
+    List<SyntaxError> errors = new ArrayList<>();
+    validatePicClauseUsage().unwrap(errors::addAll);
+    validateCompatibleUsageClause().unwrap(errors::addAll);
```

Comment:
The result of calling these methods not used, so they may return lists of errors

Reply:
Done